appserver/payara-appserver-modules/healthcheck-checker/src/main/java/fish/payara/healthcheck/mphealth/MicroProfileHealthChecker.java

```diff
@@ -115,44 +120,58 @@
     @Inject
     private PayaraInstanceImpl payaraMicro;
 
+    private final Map<String, Integer> serverHttpStatus = new ConcurrentHashMap<>();
+
+    @Override
+    public void collect(MonitoringDataCollector collector) {
+        if (isReady()) {
+            MonitoringDataCollector statusCollector = collector.in("health-check").type("checker").entity("MP")
+                    .collect("checksDone", getChecksDone())
+                    .collectNonZero("checksFailed", getChecksFailed());
+            for (Entry<String, Integer> status : serverHttpStatus.entrySet()) {
+                statusCollector.tag("server", status.getKey()).collect("statusCode", status.getValue());
+            }
+        }
+    }
+
     @Override
     public void postConstruct() {
         postConstruct(this, MicroProfileHealthCheckerConfiguration.class);
     }
 
     @Override
-    public HealthCheckResult doCheck() {
+    protected HealthCheckResult doCheckInternal() {
         HealthCheckResult result = new HealthCheckResult();
 
         if (!envrionment.isDas()) {
             //currrently this should only run on DAS
             return result;
         }
+        serverHttpStatus.clear();
 
         Map<String, Future<ClusterCommandResult>> configs = payaraMicro.executeClusteredASAdmin(GET_MP_CONFIG_STRING, new String[0]);
 
-        HashSet<String> usedInstances = new HashSet<String>();
-        
+        HashSet<String> usedInstances = new HashSet<>();
+
         //get all instances that this server knows about
         for (Server server : domain.getServers().getServer()) {
-            
-            usedInstances.add(server.getName());
+
+            @Pattern(regexp = "[A-Za-z0-9_][A-Za-z0-9\\-_\\.;]*", message = "{server.invalid.name}", payload = Server.class)
+            String instanceName = server.getName();
+            usedInstances.add(instanceName);
 
             Future taskResult = payaraExecutorService.submit(() -> {
 
                 //get the remote server's MP HealthCheck config
                 MetricsHealthCheckConfiguration metricsConfig = server.getConfig().getExtensionByType(MetricsHealthCheckConfiguration.class);
                 if (metricsConfig != null && Boolean.valueOf(metricsConfig.getEnabled())) {
                     try {
-                        String endpoint = metricsConfig.getEndpoint();
-
-                        URI remote = buildURI(server, endpoint);
-
-                        result.add(pingHealthEndpoint(remote));
-
+                        result.add(pingHealthEndpoint(instanceName,  buildURI(server, metricsConfig.getEndpoint())));
                     } catch (URISyntaxException ex) {
+                        serverHttpStatus.put(instanceName, 420);
```

Comment:
Constant may be defined for `420` status code as it is not part of the HTTP standard.

Reply:
Actually I think I should change it to a 5xx code. Both cases I used it it is a server error more then a client error.

Reply:
Only one really applying is plain 500 so I changed to that.